### ISSUE TITLE
Accessing FeedResponse.ResponseHeaders throws NullReferenceException (.net core)

### ISSUE THREAD

**Issue by bkaidbb** [Azure/azure-cosmos-dotnet-v2#612]
Package tested: Microsoft.Azure.DocumentDB.Core
Versions tested: 2.0.0, 2.1.2
OS'es tested: Windows 10, Docker Nano Server

Code sample:
```
var documentClient = new DocumentClient(new Uri("**serviceEndpoint**"), "**accessKey**");
var query = documentClient.CreateDocumentQuery(
    UriFactory.CreateDocumentCollectionUri("**dbName**", "**collectionName**"),
                "SELECT TOP 1 * FROM c",
                new FeedOptions
                {
                    PopulateQueryMetrics = true,
                    MaxItemCount = -1,
                    MaxDegreeOfParallelism = -1,
                    EnableCrossPartitionQuery = true
                }).AsDocumentQuery();

var result = await query.ExecuteNextAsync();
var headers = result.ResponseHeaders;
```

Expected behavior: 
Can access response headers

Actual behavior:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Azure.Documents.Collections.DictionaryNameValueCollection.ToNameValueCollection()
```

**Comment by srinathnarayanan** [comment]
Hi @bkaidbb Can you please update the incident you created with us, with a repro console app? Thanks!

**Comment by bkaidbb** [comment]
@srinathnarayanan here you go:

ConsoleApp1.csproj:
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <LangVersion>latest</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="2.1.2" />
  </ItemGroup>

</Project>
```

Program.cs: 
```
using System;
using System.Threading.Tasks;
using Microsoft.Azure.Documents.Client;
using Microsoft.Azure.Documents.Linq;

namespace ConsoleApp1
{
    class Program
    {
        static async Task Main(string[] args)
        {
            var documentClient = new DocumentClient(new Uri("https://**accountName**.documents.azure.com:443/"), "**accountKey**");
            var query = documentClient.CreateDocumentQuery(
                UriFactory.CreateDocumentCollectionUri("**dbName**", "**collectionName**"),
                "SELECT TOP 1 * FROM c",
                new FeedOptions
                {
                    PopulateQueryMetrics = true,
                    MaxItemCount = -1,
                    MaxDegreeOfParallelism = -1,
                    EnableCrossPartitionQuery = true
                }).AsDocumentQuery();

            var result = await query.ExecuteNextAsync();
            var headers = result.ResponseHeaders;

            Console.WriteLine("Press any key to exit...");
            Console.ReadKey();
        }
    }
}
```

**Comment by srinathnarayanan** [comment]
I tried reproing this issue on a private account and was unable to. Can you please send the console app code (with your account endpoint and read key) as part of the incident you created with CosmosDB? Thanks!

**Comment by bkaidbb** [comment]
This issue is fixed in [2.2.0 release](https://www.nuget.org/packages/Microsoft.Azure.DocumentDB.Core/2.2.0)
